packages/blockfrost/src/blockfrostProvider.ts

```diff
@@ -66,6 +66,35 @@ const toProviderError = (error: unknown) => {
 export const blockfrostProvider = (options: Options, logger = dummyLogger): WalletProvider => {
   const blockfrost = new BlockFrostAPI(options);
 
+  const fetchByAddressSequentially = async <Item, Response>(
```

Comment:
Nice function. Could be made even more generic and used for non-address queries, e.g. `const fetchSequentially = async <Item, Arg, Response>(`. Not a requested change for this PR, just a note.

Reply:
I almost went to that design, but decided to keep it address-specific until we need more capabilities.